samza-azure/src/main/java/org/apache/samza/system/azureblob/utils/BlobMetadataContext.java

```diff
@@ -25,10 +25,12 @@
 public class BlobMetadataContext {
   private final String streamName;
   private final long blobSize;
+  private final long numberOfMessagesInBlob;
 
-  public BlobMetadataContext(String streamName, long blobSize) {
+  public BlobMetadataContext(String streamName, long blobSize, long numberOfMessagesInBlob) {
     this.streamName = streamName;
     this.blobSize = blobSize;
+    this.numberOfMessagesInBlob = numberOfMessagesInBlob;
```

Comment:
What will this be used for or what do we anticipate it will be used for?

Reply:
This value will be given to the metadata generator Impl passed in by the user of the SystemProducer through a config (https://github.com/apache/samza/pull/1358).  

So along with streamName, blobSize, this numberOfMessagesInBlob is anticipated to be used to build metadata properties of the blob created. Metadata proprties are kv pairs attached to the blob that can then be leveraged for kusto ingestion. So I expect the usage to be something like <stream:streamName, rawSize: blobSize, records: numberofMessages>

Reply:
I see. Thanks for explaining.

I notice that the interface of the generator takes a BlobMetadataContext and returns a Map<String,String>; out of curiosity, is it intended that a generator may have additional metadata (beyond what is carried within the BlobMetadataContext) to attach to the blob?